### ISSUE TITLE
User doesn't exist even though it does

### ISSUE THREAD

**Issue by jakubgs** [webiny/webiny-js#868]
## Bug

I've created a user via my admin account with full permissions but when trying to log in I see the "User doesn't exist" error like so:
![webiny_no_user](https://user-images.githubusercontent.com/2212681/82576642-34b09000-9b8a-11ea-9488-114d33dd21db.png)
But that's clearly not true because I can see the user in the Admin UI and in MongoDB:
```
webiny:PRIMARY> db.SecurityUser.findOne({email:"joe@example.org"})
{
	"_id" : ObjectId("5ec6952b9fbd570008255127"),
	"id" : "5ec6952b6549b4000863393d",
	"createdBy" : "5ec6937d46627e0008b4e62e",
	"deleted" : false,
	"createdOn" : ISODate("2020-05-21T14:50:19.349Z"),
	"savedOn" : ISODate("2020-05-21T14:50:19.309Z"),
	"email" : "joe@example.org",
	"password" : "XYZ",
	"firstName" : "Joe",
	"lastName" : "Smith"
}
```

## Specifications

* Version: `3.2.8`

## Expected Behavior
User can log in after being created.

## Actual Behavior
User cannot log in, even after waiting quite a bit.

## Steps to Reproduce the Problem
It's just a fresh `prod` env installation on AWS.

## Detailed Description
I didn't change anything. I just provided a `.env.json` with the right AWS profile and MongoDB URI.
I've checked the CloudWatch logs but couldn't find any errors in them.

**Comment by Pavel910** [comment]
Hi @jakubgs, thanks for reporting! We'll investigate this and get back to you :)

**Comment by jakubgs** [comment]
Happy to help you debug, I still have the infra up, so I can check things that you suggest.

**Comment by Pavel910** [comment]
@jakubgs just tried this, and the problem is that although the user does exist in the Database, it doesn't get created in the Cognito User Pool. When you go to login form, and try to login, first the browser communicates with Cognito - and it tells you that the user doesn't exist.

I'll investigate this further and release a fix when I figure out the problem.

Once again, thanks for reporting!

**Comment by Pavel910** [comment]
@jakubgs sorry for taking this long, we were overwhelmed by preparations for the new launch. Also just this morning we managed to figure out what the problem is thanks to several of our community members: just use password at least 8 characters long 😆 

What an oversight on our side 🤦 We'll handle the validation on that password field ASAP.

Cheers!

**Comment by jakubgs** [comment]
That's great to hear, thanks man.

**Comment by jakubgs** [comment]
Nice, I assume this will be released with `v4.0.3`, right? When is that coming?

**Comment by Pavel910** [comment]
@jakubgs `4.1.0` coming during tomorrow :)
